File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        insertI = bisect_left(intervals, newInterval)
        intervals.insert(insertI, newInterval)
        
        stack = []
        
        for s, e in intervals:
            if stack and stack[-1][1] >= s:
                lastS, lastE = stack.pop()
                stack.append([lastS, max(lastE, e)])
            else:
                stack.append([s, e])
        return stack

File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int position = upper_bound(intervals.begin(), intervals.end(), newInterval) - intervals.begin();
        intervals.insert(intervals.begin() + position, newInterval);
        vector<vector<int>> answer;
        
        for (int i = 0; i < intervals.size(); ++i){
            if (answer.empty() || answer.back()[1] < intervals[i][0]) answer.push_back(intervals[i]);
            else answer.back()[1] = max(answer.back()[1], intervals[i][1]);
        }
        return answer;
    }
};